tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js

```diff
@@ -0,0 +1,62 @@
+import { singularize } from 'ember-inflector';
+import { isPathParam, removePrefix, isSingleResourceIdentifier } from './resourceEvaluation.js';
+import { casing } from '@stoplight/spectral-functions';
+
+/**
+ * Returns IPA Compliant Operation ID.
+ *
+ * @param method the standard method name (create, update, get etc.), custom method name, or empty string (only for legacy custom methods)
+ * @param path the path for the endpoint
+ */
+export function generateOperationID(method, path) {
+  let resourceIdentifier = removePrefix(path);
+  let nouns = resourceIdentifier
+    .split('/')
+    .filter((id) => id.length > 0 && !isPathParam(id))
```

Comment:
```suggestion
    .filter((section) => section.length > 0 && !isPathParam(section))
```
[Nit] `section` or `segment` on naming to clarify that these are not necessarily IDs/path params